core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -661,18 +745,21 @@ final class ShuffleBlockFetcherIterator(
       result match {
         case r @ SuccessFetchResult(blockId, mapIndex, address, size, buf, isNetworkReqDone) =>
           if (address != blockManager.blockManagerId) {
-            if (hostLocalBlocks.contains(blockId -> mapIndex)) {
-              shuffleMetrics.incLocalBlocksFetched(1)
-              shuffleMetrics.incLocalBytesRead(buf.size)
-            } else {
-              numBlocksInFlightPerAddress(address) = numBlocksInFlightPerAddress(address) - 1
-              shuffleMetrics.incRemoteBytesRead(buf.size)
-              if (buf.isInstanceOf[FileSegmentManagedBuffer]) {
-                shuffleMetrics.incRemoteBytesReadToDisk(buf.size)
-              }
-              shuffleMetrics.incRemoteBlocksFetched(1)
-              bytesInFlight -= size
-            }
+           if (hostLocalBlocks.contains(blockId -> mapIndex) ||
+             pushBasedFetchHelper.isLocalPushMergedBlockAddress(address)) {
+             // It is a host local block or a local shuffle chunk
+             shuffleMetrics.incLocalBlocksFetched(1)
+             shuffleMetrics.incLocalBytesRead(buf.size)
+           } else {
+             // Could be a remote shuffle chunk or remote block
+             numBlocksInFlightPerAddress(address) = numBlocksInFlightPerAddress(address) - 1
+             shuffleMetrics.incRemoteBytesRead(buf.size)
+             if (buf.isInstanceOf[FileSegmentManagedBuffer]) {
+               shuffleMetrics.incRemoteBytesReadToDisk(buf.size)
+             }
+             shuffleMetrics.incRemoteBlocksFetched(1)
+             bytesInFlight -= size
+           }
```

Comment:
Most of this diff is due to change in indentation - revert back to 2 space ?

Reply:
Actually @mridulm. I didn't even change the indentation here. It same as before. I just added lines 749, 750, and 754. I will remove the comment at 754 and see if this interface stops showing that I have changed this.

Reply:
This is fixed now.